src/effects/DepthOfField.tsx

```diff
@@ -19,21 +19,80 @@ export const DepthOfField = forwardRef(function DepthOfField(
   ref: Ref<DepthOfFieldEffect>
 ) {
   const invalidate = useThree((state) => state.invalidate)
-  const { camera } = useContext(EffectComposerContext)
+  const { camera, scene } = useContext(EffectComposerContext)
   const effect = useMemo(() => {
     const effect = new DepthOfFieldEffect(camera, props)
     // Temporary fix that restores DOF 6.21.3 behavior, everything since then lets shapes leak through the blur
     const maskMaterial = (effect as any).maskPass.getFullscreenMaterial()
     maskMaterial.maskFunction = MaskFunction.MULTIPLY_RGB_SET_ALPHA
     return effect
-  }, [camera, props])
+  }, [])
+
+  useLayoutEffect(() => {
+    return () => {
+      const depthEffect = effect as any
+      depthEffect.blurPass.renderTargetA.dispose()
+      depthEffect.blurPass.renderTargetB.dispose()
+      depthEffect.maskPass.dispose()
+      depthEffect.cocPass.dispose()
+      depthEffect.renderTarget.dispose()
+      depthEffect.renderTargetCoC.dispose()
+      depthEffect.renderTargetCoCBlurred.dispose()
+      depthEffect.renderTargetMasked.dispose()
+      depthEffect.renderTargetNear.dispose()
+      depthEffect.renderTargetFar.dispose()
+    }
+  }, [])
```

Comment:
Can you try `depthEffect.dispose()` here instead in a `useEffect` and populate the dependency array? This handles the above edge cases https://github.com/pmndrs/postprocessing/blob/main/src/effects/Effect.js#L454, and should outright fix the issue.

```js
useEffect(() => {
  return () => {
    effect.dispose()
  }
}, [effect])
```